src/squlearn/qnn/lowlevel_qnn_pennylane.py

```diff
@@ -0,0 +1,980 @@
+            else:
+                params.update(self._observable.get_params())
+        return params
+
+    @property
+    def num_qubits(self) -> int:
+        """Return the number of qubits of the QNN"""
+        return self._pennylane_circuit._num_qubits
+
+    @property
+    def num_features(self) -> int:
+        """Return the dimension of the features of the PQC"""
+        return self._pqc.num_features
+
+    @property
+    def num_parameters(self) -> int:
+        """Return the number of trainable parameters of the PQC"""
+        return self._pqc.num_parameters
+
+    @property
+    def num_operator(self) -> int:
+        """Return the number outputs"""
+        return self._num_operators
+
+    @property
+    def num_parameters_observable(self) -> int:
+        """Return the number of trainable parameters of the expectation value operator"""
+        return self._num_parameters_observable
+
+    @property
+    def multiple_output(self) -> bool:
+        """Return true if multiple outputs are used"""
+        return self._multiple_output
+
+    @property
+    def parameters(self) -> ParameterVector:
+        """Return the parameter vector of the PQC."""
+        return self._param
+
+    @property
+    def features(self) -> ParameterVector:
+        """Return the feature vector of the PQC."""
+        return self._x
+
+    @property
+    def parameters_operator(self) -> ParameterVector:
+        """Return the parameter vector of the cost operator."""
+        return self._param_obs
+
+    def evaluate(
+        self,
+        x: Union[float, np.ndarray],
+        param: Union[float, np.ndarray],
+        param_obs: Union[float, np.ndarray],
+        *values: Union[
+            str,
+            direct_evaluation,
+            post_processing_evaluation,
+            ParameterVector,
+            ParameterVectorElement,
+            tuple,
+        ],
+    ) -> dict:
+        """General function for evaluating the output of derivatives of the QNN.
+
+        Evaluation works for given combination of
+        input features `x` and parameters `param` and `param_op`.
+        The function includes caching of results
+
+        If `x`, `param`, and/or `param_op` are given as a nested list
+        (for example multiple sets of parameters),
+        the values are returned in a nested list.
+
+        Args:
+            x (np.ndarray): Values of the input feature data.
+            param (np.ndarray): Parameter values of the PQC parameters
+            param_op (np.ndarray): Parameter values of the operator parameters
+            values : Derivatives (or values) of the QNN that are evaluated. Higher order
+                     derivatives are given as tuples of parameters or vectors.
+
+        Results:
+            Returns a dictionary with the computed values.
+            The keys of the dictionary are given by the entries in the values tuple
+
+        """
+
+        def _evaluate_todo_single_x(
```

Comment:
also here, why is this indented?

Reply:
Don't get it, that is a helperfunction only need in evaluate

Reply:
Don